packages/flutter/lib/src/services/raw_keyboard_linux.dart

```diff
@@ -19,16 +19,16 @@ import 'raw_keyboard.dart';
 class RawKeyEventDataLinux extends RawKeyEventData {
   /// Creates a key event data structure specific for macOS.
   ///
-  /// The [toolkit], [scanCode], [codePoint], [keyCode], and [modifiers], arguments
-  /// must not be null.
+  /// The [toolkit], [scanCode], [charactersIgnoringModifiers], [keyCode], and
+  /// [modifiers], arguments must not be null.
   const RawKeyEventDataLinux({
     @required this.keyHelper,
+    this.charactersIgnoringModifiers = '',
```

Comment:
Should this be `characterIncludingModifiers` (`character` singular)? The doc comment below describes this as "the name of the printable key" -- is this something like `e`, `E`, `é`, `¸` (cédille), `return` (for 0x0d)? 

I wonder if there's a way to avoid the use of the word 'character' which is pretty ambiguous when you get down to the level of differentiating between codepoints, glyphs, bytes and other things frequently referred to as 'characters' in discussions at this level.

Reply:
I got that name from the macos implementation, which is the name Apple uses for their own API. So maybe it's good to keep for consistency? I agree that it being plural makes it a bit more ambiguous since it can be one character, but two codepoints. I could make it singular, but don't really have any other suggestions to change the word "character" :) WDYT?

Reply:
On a second thought, given that the macOS implementation has the same name, I'd like to keep the same name for consistency. WDYT?

Reply:
I agree it's a good idea to keep ourselves consistent across platforms, but Flutter's APi doesn't need to necessarily leak macOS concepts into our own APIs. To be honest, I'm still confused as to which meaning of 'character' this refers to and what that would look like in the presence of e.g. an IME, even after reading [Apple's docs](https://developer.apple.com/documentation/appkit/nsevent/1524605-charactersignoringmodifiers?language=objc). I'd guess (but am not at all sure), that a keypress of the 'c' key when using a Japanese keyboard layout in 'romaji' input mode would emit 'c', but that in 'kana' input mode it might emit 'そ' (see my profile picture).

In any case, depending on what the behaviour is, I wonder if we can find a better name to expose than 'character'. @Hixie almost certainly has thoughts on this.

In particular, as noted above the word 'character' is really ambiguous and feels like a leftover from the old days where byte/grapheme/glyph/codepoint were regularly conflated, and APIs that don't communicate concepts precisely to users result in assumptions + bugs.

Reply:
Definitely don't use the word "character". It's a meaningless word.